src/main/java/seedu/contax/logic/parser/RangeCommandParser.java

```diff
@@ -31,13 +30,20 @@ public RangeCommand parse(String args) throws ParseException {
         Index toIndex;
 
         if (argMultimap.getValue(PREFIX_RANGE_FROM).isPresent() && argMultimap.getValue(PREFIX_RANGE_TO).isPresent()) {
-            fromIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_FROM).get()));
-            toIndex = Index.fromOneBased(Integer.parseInt(argMultimap.getValue(PREFIX_RANGE_TO).get()));
+            try {
+                fromIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_RANGE_FROM).get());
+                toIndex = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_RANGE_TO).get());
+            } catch (ParseException pe) {
+                throw new ParseException(String.format(MESSAGE_INVALID_RANGE_INDEX,
+                        RangeCommand.MESSAGE_USAGE), pe);
```

Comment:
I think you should explicitly say that the index is invalid, instead of just displaying the syntax.

Reply:
thx changed